sunpy/map/tests/test_mapbase.py

```diff
@@ -821,15 +821,78 @@ def test_quicklook(aia171_test_map):
         assert aia171_test_map._repr_html_() in html_string
 
 
-def test_submap_non_kwarg_parameters(aia171_test_map):
-    bl = SkyCoord(-300*u.arcsec, -300*u.arcsec, frame=aia171_test_map.coordinate_frame)
-    tr = SkyCoord(500*u.arcsec, 500*u.arcsec, frame=aia171_test_map.coordinate_frame)
+def test_submap_inputs(generic_map):
+    bl_pix = [3, 2] * u.pix
+    bl_coord = SkyCoord(20, -10, unit=u.arcsec,
+                        frame=generic_map.coordinate_frame)
+
+    tr_pix = [5, 4] * u.pix
+    tr_coord = SkyCoord(0, 10, unit=u.arcsec,
+                        frame=generic_map.coordinate_frame)
+
+    bl_tr_coord = SkyCoord([20, 0], [-10, 10], unit=u.arcsec,
+                           frame=generic_map.coordinate_frame)
+
+    width_pix = 2 * u.pix
+    height_pix = 2 * u.pix
+    width_deg = 20 * u.arcsec
+    height_deg = 20 * u.arcsec
+
+    # works
+    smap = generic_map.submap(bl_coord, top_right=tr_coord)
+    # assert u.allclose(smap.dimensions, (1, 1) * u.pix)
+    smap = generic_map.submap(bl_coord, width=width_deg, height=height_deg)
+    # assert u.allclose(smap.dimensions, (1, 1) * u.pix)
+    smap = generic_map.submap(bl_tr_coord)
+    # assert u.allclose(smap.dimensions, (1, 1) * u.pix)
+
+    # works
+    smap = generic_map.submap(bl_pix, top_right=tr_pix)
+    # assert u.allclose(smap.dimensions, (1, 1) * u.pix)
+    # smap = generic_map.submap(bl_pix, width=width_pix, height=height_pix)
```

Comment:
This errors and should pass.

Reply:
This is resolved by making the following modifications:
```
        elif isinstance(bottom_left, u.Quantity) and bottom_left.unit.is_equivalent(u.pix):

            if isinstance(top_right, u.Quantity) and top_right.unit.is_equivalent(u.pix):

                if len(bottom_left) != 2 or len(top_right) != 2:
                    raise TypeError("bottom_left and top_right in pixels must contain both coordinates of the pixel")

                x_pixels = u.Quantity([bottom_left[0], top_right[0]]).value
                y_pixels = u.Quantity([top_right[1], bottom_left[1]]).value

            elif (isinstance(width, u.Quantity) and width.unit.is_equivalent(u.pix) and
                  isinstance(height, u.Quantity) and height.unit.is_equivalent(u.pix)):

                x_pixels = u.Quantity([bottom_left[0], bottom_left[0] + width]).value
                y_pixels = u.Quantity([bottom_left[1], bottom_left[1] + height]).value

```